esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -1858,6 +1871,8 @@ void doWUQueryWithSort(IEspContext &context, IEspWUQueryRequest & req, IEspWUQue
     addWUQueryFilter(filters, filterCount, filterbuf, req.getJobname(), (WUSortField) (WUSFjob | WUSFnocase));
     addWUQueryFilter(filters, filterCount, filterbuf, req.getECL(), (WUSortField) (WUSFecl | WUSFwild));
 
+    addWUQueryFilterTotalClusterTime(sortorder, filters, filterCount, filterbuf, req.getTotalClusterTimeThresholdMilliSec(), WUSFtotalthortime);
```

Comment:
@wangkx - it looks to me, that if there is any other sort order read above (via readWUQuerySortOrder), then it will be lost in favour of WUSFtotalthortime.
Is that correct, and isn't that confusing from a UI perspective vs other fields?

At a minimum I think needs some comments here as to why this field is treated differently from others.

Reply:
> if there is any other sort order read above (via readWUQuerySortOrder), then it will be lost in favour of WUSFtotalthortime. Is that correct, and isn't that confusing from a UI perspective vs other fields?
Yes. It is correct. UI may have to add some text around the Input box of a FilterByTotalThorTime. 
>At a minimum I think needs some comments here as to why this field is treated differently from others.
There is a comment at the line 1763.

Reply:
I guess my question was:
>//Cassandra WUs can be filtered by TotalClusterTime only if sorting by TotalClusterTime.

Why is that when not for other fields?

Reply:
>Yes. It is correct. UI may have to add some text around the Input box of a FilterByTotalThorTime.

If the UI needs changes to clarify loss of sort order, is there a separate JIRA?
It's still not obvious from reading the code though, so there should be commend explaining the logic, and reasons behind it.

Reply:
> Why is that when not for other fields?
In fact, I am not familiar with CassandraWU code. I have not checked how other sort goes. 
 
> If the UI needs changes to clarify loss of sort order, is there a separate JIRA?
There is a JIRA for UI.

Reply:
HPCC-23098 ? 

I might be misunderstanding.
When you filter by total cluster time, does it effect the display sort order?
That's going to be confusing if so, and it's not clear (to me) yet why (and not necessary in Dali case). And not mentioned in HPCC-23098 either.

Reply:
Yes HPCC-23098. I haven't mentioned this behavior there because I am not sure we will accept this behavior (but, I did mention it with both Miguel and Kanul). It will be clear if this PR is accepted.

> When you filter by total cluster time, does it effect the display sort order?

Yes. this is the behavior I am talking about.